### PR TITLE
Cancel operation cash payment

### PR DESCRIPTION
**Pull request by lomamech**
-  Implement client side for Cancel Operation Cash Payment
- Update the showing of cancelled payment
-Update service checkCashPayment for the checking of payment Invoice 
Update invoice registry for warning if the invoice is yet paid, during
credit note operation
Update api for cash for the request of invoice paid
Update sql request for the research of patient invoice
- Implementation of integration test
-	Fix error arround procedures.sql  
Update configuration data'test in data.sql
Fix error in data.sql arround aged debtor
Update test journal with the new configuration data

@jniles  @sfount  @mbayopanda

### CODE REVIEW

File: client/src/js/services/CashService.js
Comment (jniles) [inline client/src/js/services/CashService.js:30]: This is the correct place to put these.  Nice!

File: client/src/js/services/CashService.js
Comment (jniles) [inline client/src/js/services/CashService.js]: Should this be in `finance/reports`?   I think it should probably be in `partials/cash/`

File: client/src/js/services/CashService.js
Comment (jniles) [inline client/src/js/services/CashService.js]: What is the second parameter `true` for?

File: client/src/partials/patient_invoice/registry/registry.js
Comment (jniles) [inline client/src/partials/patient_invoice/registry/registry.js]: Can this not use a uuid?  It would be much more secure.

File: server/controllers/finance/debtors/index.js
Comment (jniles) [inline server/controllers/finance/debtors/index.js:263]: :+1:

File: server/controllers/finance/cash.js
Comment (jniles) [inline server/controllers/finance/cash.js]: This could use a better name.  Maybe checkReversedPayment?  Or checkInvoicePayment?

File: server/controllers/finance/cash.js
Comment (jniles) [inline server/controllers/finance/cash.js]: What is `voucher.type_id = 10`?

File: server/controllers/finance/reports/financial.patient.handlebars
Comment (jniles) [inline server/controllers/finance/reports/financial.patient.handlebars:43]: In the future, it would be better to use a descriptive class name.  This is not a huge problem though.

File: server/models/test/data.sql
Comment (jniles) [inline server/models/test/data.sql:273]: :+1:

File: test/end-to-end/cash/cash.registry.spec.js
Comment (jniles) [inline test/end-to-end/cash/cash.registry.spec.js:14]: Is this the cash registry?  Should it be called cash registry?

**Review by jniles — COMMENTED** [review]
@lomamech, I've completed a review.  The code looks pretty good!  I've made a few comments, mostly on naming things and best practice.  After that, I think we can probably merge this.

Good work! :+1:

**Comment by lomamech** [comment]
Complete all reviews 

Update files location move in client/src/partials/cash/modals
Update parameters for cashServices using only invoiceUuid
Update the name of function chekin
Complete translate file fr.json

@jniles

File: server/controllers/finance/cash.js
Comment (lomamech) [reply server/controllers/finance/cash.js]: Voucher 10 is used as being the entity_type value when it comes to a transaction reverse

**Comment by jniles** [comment]
@lomamech, thanks!  There is one conflict - can you rebase with master?

**Comment by lomamech** [comment]
Fix, sanitaze and update some data for integration and end to end test

@jniles

File: server/models/procedures.sql
Comment (jniles) [inline server/models/procedures.sql:646]: I think this is wrong.  However, I'll let @sfount review it.

@sfount, can  you make sure this posting work is correct?

Comment (sfount) [reply server/models/procedures.sql:646]: @jniles @lomamech Why are the comments I made removed? It looks to me like this was a problem with merging or rebasing branches.

As the removed comment points out writing this line causes the debt to the debtor to not balance. This will cause `master` to create inbalanced transactions. 

Please review the latest `master` file [procedures.sql](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/models/procedures.sql) to see what should be kept and removed here.

**Review by sfount — CHANGES_REQUESTED** [review]
It looks like there has been a problem with merging/ rebasing to master with this branch. Currently the `PostCash` procedure will cause imbalanced transactions. This should be addressed before merging with `master`.

**Comment by lomamech** [comment]
I have remake this work in another pull request

**Comment by jniles** [comment]
Cool :+1:

For reference: #941.

**Pull request closed without merging**
